packages/eslint-plugin/src/rules/no-unnecessary-type-assertion.ts

```diff
@@ -48,6 +49,10 @@ export default createRule<Options, MessageIds>({
         type: 'object',
         additionalProperties: false,
         properties: {
+          ignoreStringConstAssertion: {
```

Comment:
[Bug] From https://github.com/typescript-eslint/typescript-eslint/issues/8721#issuecomment-2455320478, I believe the resolution was to have this new option be on by default. I think this should instead be named, maybe, `check*ConstAssertions` (`*` pending based on the other naming comment).

Reply:
Oh I understood the contrary and was willing to be conservative but if project is willing to lax the rule by default I'm all in (less configuration on my side 😉)